### PR TITLE
Revert BC breaking change in PennTreebank dataset

### PR DESCRIPTION
**Pull request by parmeet**
Revert the BC breaking change in the output introduced in https://github.com/pytorch/text/pull/1511

### CODE REVIEW

**Comment by erip** [comment]
LGTM! Qq, did tests not catch this?

**Comment by parmeet** [comment]
> LGTM! Qq, did tests not catch this?

ya, they did, I learnt from there only for eg: https://app.circleci.com/pipelines/github/pytorch/text/4693/workflows/447cb44d-4ee2-4361-b954-2eebe700f25d/jobs/157132. Unfortunately, earlier the cache was broken though :(

File: torchtext/datasets/penntreebank.py [hunk 1]
```diff
@@ -50,4 +50,4 @@ def PennTreebank(root, split: Union[Tuple[str], str]):
     cache_dp = HttpReader(cache_dp).end_caching(mode="wb", same_filepath_fn=True)
     data_dp = FileOpener(cache_dp, mode="r")
     # remove single leading and trailing space from the dataset
-    return data_dp.readlines(return_path=False).map(lambda t: t.strip())
+    return data_dp.readlines(return_path=False, strip_newline=False)
```
Comment (abhinavarora) [inline torchtext/datasets/penntreebank.py:53]: Same comment as #1559

**Comment by abhinavarora** [comment]
> > LGTM! Qq, did tests not catch this?
> 
> ya, they did, I learnt from there only for eg: https://app.circleci.com/pipelines/github/pytorch/text/4693/workflows/447cb44d-4ee2-4361-b954-2eebe700f25d/jobs/157132. Unfortunately, earlier the cache was broken though :(

Wondering why did local testing not catch it?

**Comment by parmeet** [comment]
closing this PR as this is a reasonable BC breaking change.

**Pull request closed without merging**